plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -347,8 +355,88 @@ private function handle_data_sync_option_changed( string $feature_id ) {
 		// some reason it was ongoing while it was disabled).
 		if ( $data_sync_is_enabled ) {
 			$this->batch_processing_controller->enqueue_processor( DataSynchronizer::class );
+			$this->maybe_schedule_pending_sync_check();
 		} else {
 			$this->batch_processing_controller->remove_processor( DataSynchronizer::class );
+			$this->maybe_unschedule_pending_sync_check();
+		}
+	}
+
+	/**
+	 * Schedule a recurring event to check for pending order syncs.
+	 *
+	 * Wrapper to avoid throwing an error if Action Scheduler isn't initialized yet.
+	 *
+	 * @return void
+	 */
+	private function maybe_schedule_pending_sync_check() {
+		if ( ! ActionScheduler::is_initialized() ) {
+			return;
+		}
+
+		$has_scheduled_action = WC()->queue()->search( array(
+			'hook'   => self::SYNC_CHECK_EVENT_HOOK,
+			'status' => array( 'in-progress', 'pending' ),
+		) );
+
+		if ( ! $has_scheduled_action ) {
+			WC()->queue()->schedule_recurring(
+				time() + HOUR_IN_SECONDS,
+				6 * HOUR_IN_SECONDS,
+				self::SYNC_CHECK_EVENT_HOOK
+			);
+		}
+	}
+
+	/**
+	 * Remove any previously scheduled events to check for pending order syncs.
+	 *
+	 * Wrapper to avoid throwing an error if Action Scheduler isn't initialized yet.
+	 *
+	 * @return void
+	 */
+	private function maybe_unschedule_pending_sync_check() {
+		if ( ! ActionScheduler::is_initialized() ) {
+			return;
+		}
+
+		WC()->queue()->cancel_all( self::SYNC_CHECK_EVENT_HOOK );
+	}
+
+	/**
+	 * When sync is turned on, check to see if there are any pending syncs and if so, queue the processor.
+	 *
+	 * Runs as a recurring callback for SYNC_CHECK_EVENT_HOOK.
+	 *
+	 * @return void
+	 */
+	private function check_for_pending_syncs() {
+		if ( ! $this->data_synchronizer->data_sync_is_enabled() ) {
+			$this->maybe_unschedule_pending_sync_check();
+			return;
+		}
+
+		$sync_status = $this->data_synchronizer->get_sync_status();
```

Comment:
instead of getting_sync_status, get pending sync count would be more accurate. sync status can be outdated if there are new orders after the last time status was updated (for example, between multiple sync action runs)

Reply:
Ah thanks, I didn't realize it was a cached value. Fixed in 9c96a84438